### PR TITLE
ENH: Add SH order theory

### PR DESCRIPTION
**Pull request by kaitj**
Related to #28 - not entirely sure what you had in mind here. Added brief mention of how the sh order is related to the angular frequency, but I think as it is, it currently provides the relevant details for an introductory lesson.

### CODE REVIEW

File: _episodes/constrained_spherical_deconvolution.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ In order to perform the deconvolution over the sphere, the spherical
 representation of the diffusion data has to be obtained. This is done using the
 so-called Spherical Harmonics (SH) which are a basis that allow to represent
 any function on the sphere (much like the Fourier analysis allows to represent
-a function in terms of in terms of trigonometric functions).
+a function in terms cartesian coordinates).
```
Comment (jhlegarreta) [inline _episodes/constrained_spherical_deconvolution.md:44]: I'd say that the function can be represented in any coordinate system one chooses, not necessarily Cartesian (e.g. can be polar, spherical, etc.). To me the relevant point was that the FT represents a signal in terms of trigonometric functions in a similar way that functions on the sphere allow for a representation using SH functions.

File: _episodes/constrained_spherical_deconvolution.md [hunk 2]
```diff
@@ -218,7 +219,7 @@ where $R$ is the number of coefficients. For example, an order $l_{max} = {4, 6,
 series has $R = {15, 28, 45}$ coefficients, respectively. Note the use of even
 orders: even order SH functions allow to reconstruct symmetric spherical
 functions. Traditionally, even orders have been used motivated by the fact that
-the diffusion process is symmetric around the origin.
+the diffusion process is symmetric around the origin.
```
Comment (jhlegarreta) [inline _episodes/constrained_spherical_deconvolution.md:222]: Please remove the end of line whitespace.

File: _episodes/constrained_spherical_deconvolution.md [hunk 3]
```diff
@@ -209,7 +209,8 @@ and hence it must be computed on a case basis.
 After estimating a response function, the fODF is reconstructed through the
 deconvolution operation. In order to obtain the spherical representation of the
 diffusion signal, the order of the Spherical Harmonics expansion must be
-specified. The series is infinite, but must be truncated to a maximum order in
+specified. The order, $l$, corresponds to an angular frequency of the basis function. 
+While, the series is infinite, it must be truncated to a maximum order in
```
Comment (jhlegarreta) [inline _episodes/constrained_spherical_deconvolution.md:213]: Unnecessary comma after `While`.

**Review by jhlegarreta — COMMENTED** [review]
A few in-line comments. Not all changes have been reflected in the Jupyter Notebook. Was this intentional?

**Comment by jhlegarreta** [comment]
> Related to #28 - not entirely sure what you had in mind here. A

As mentioned in #30, introducing the mathematical expressions related to the SH will greatly improve the foundations of the episode in the long run. But this is already a step forward.

File: _episodes/constrained_spherical_deconvolution.md (hunk 1 above)
Comment (kaitj) [reply _episodes/constrained_spherical_deconvolution.md:44]: That is fair, reverting this one in the upcoming commit.

File: _episodes/constrained_spherical_deconvolution.md (hunk 2 above)
Comment (kaitj) [reply _episodes/constrained_spherical_deconvolution.md:222]: Removed

**Comment by kaitj** [comment]
Agreed - I am going to priotize updating for #30 than and come back to #28 at a later date!

File: _episodes/constrained_spherical_deconvolution.md [hunk 4]
```diff
@@ -218,7 +219,7 @@ where $R$ is the number of coefficients. For example, an order $l_{max} = {4, 6,
 series has $R = {15, 28, 45}$ coefficients, respectively. Note the use of even
 orders: even order SH functions allow to reconstruct symmetric spherical
 functions. Traditionally, even orders have been used motivated by the fact that
-the diffusion process is symmetric around the origin.
+the diffusion process is symmetric around the origin.s
```
Comment (jhlegarreta) [inline _episodes/constrained_spherical_deconvolution.md:222]: Dangling after period.

Comment (kaitj) [reply _episodes/constrained_spherical_deconvolution.md]: Fixed in 83c6995

**Review by jhlegarreta — APPROVED** [review]

**Pull request merged**